server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -211,9 +213,22 @@ public VirtConsumerMap getGuestConsumersMap(Owner owner,
         // each guest ID.
 
         Query q = currentSession().createSQLQuery(sql);
-        q.setParameterList("guestids", possibleGuestIds);
-        q.setParameter("ownerid", owner.getId());
-        List<String> consumerUuids = q.list();
+        List<String> consumerUuids = new ArrayList<String>();
+
+        int fromIndex = 0;
+        int toIndex = fromIndex + MAX_IN_QUERY_LENGTH;
+
+        while (fromIndex < possibleGuestIds.size()) {
+            if (toIndex > possibleGuestIds.size()) {
+                toIndex = possibleGuestIds.size();
+            }
+            List<String> subList = possibleGuestIds.subList(fromIndex, toIndex);
+            q.setParameterList("guestids", subList);
+            q.setParameter("ownerid", owner.getId());
+            consumerUuids.addAll(q.list());
+            fromIndex = toIndex;
+            toIndex += MAX_IN_QUERY_LENGTH;
+        }
```

Comment:
Pretty sure we have a standardized way of doing this in AbstractHibernateCurator